### DISCUSSION TITLE
How to register & dispatch a event

### DISCUSSION THREAD

**Discussion by zhaoyiming0803** [microsoft/fast#6655] [Q&A] [answered]
# 🐛 Bug Report

## 😯 Current Behavior

How can i register a custom event, and dispatch it in the `my-tag`

How to receive messages that trigger by `$emit`

``` html
<my-tag onChange="onChange"></my-tag>
```

## 🌍 Your Environment

* OS & Device: [e.g. MacOS, iOS, Windows, Linux] on [iPhone 7, PC]
* Browser [e.g. Microsoft Edge, Google Chrome, Apple Safari, Mozilla FireFox]
* Version [e.g. 1.8.0]

**Comment by usrrname** [comment]
I suggested we contribute that to the docs in #6638 
If you find this helpful, please give it an upvote! 
Here's an example @zhaoyiming0803  https://stackblitz.com/edit/vitejs-vite-p2v9tm?file=README.md

**Comment by zhaoyiming0803** [comment]
The following codes can work, actually, i want to register & trigger a `custom event`:

``` html
<footer-tag slot="footer" id="footer-tag"></footer-tag>
```

``` javascript
const footerTag = document.querySelector('#footer-tag')

footerTag.addEventListener('on-change-footer-greeting', (e) => {
  console.log('on change footer greeting: ', e.detail)
})
```

My `footer-tag` Component:

``` javascript
@customElement({
  name: 'footer-tag',
  template,
  styles
})
export class Footer extends FASTElement {
   greetingChanged () {
      const customEvent = new CustomEvent('on-change-footer-greeting', {
         detail: {
            greeting: this.greeting
         }
      })
      this.dispatchEvent(customEvent)
  }
}
```

**Comment by chrisdholt** [comment, marked as answer]
Take a look at this line: https://github.com/microsoft/fast/blob/4e17af8f47fedb430ba3204269f2bca2ae413c0e/packages/web-components/fast-foundation/src/tree-item/tree-item.ts#L68

You can use the emit helper to emit a custom event with a name, and you can pass back additional information (the event object if available as an example).

FYI, I’m going to convert this to a discussion as it seems more question related than an actual issue.
